src/llama.cpp

```diff
@@ -8202,6 +8315,7 @@ static struct ggml_tensor * llm_build_kqv(
 }
 
 static struct ggml_tensor * llm_build_kv(
+       struct llama_context & lctx,
         struct ggml_context * ctx,
           const llama_model & model,
         const llama_hparams & hparams,
```

Comment:
Let's keep the `ggml_context` as the first argument to be aligned with the `ggml` interface:

```suggestion

static struct ggml_tensor * llm_build_kv(
        struct ggml_context * ctx,
       struct llama_context & lctx,
          const llama_model & model,
        const llama_hparams & hparams,
```

Also, since now we are passing the entire `llama_context`, there is no need to pass `model` and `hparams` individually - these can be obtained from `lctx`

Reply:
I removed redundant params in https://github.com/ggerganov/llama.cpp/pull/8332/commits/f68d092459059df92b0ba68b0b64282c1d56c56d

Small question: will it make sense to move `llm_build_kv`, `llm_build_kqv`, `llm_build_moe_ffn`,... to class member of `struct llm_build_context`? This way, we can prevent passing arguments like `lctx`, `kv_head`, `cb`

Reply:
Yes, I'll do a refactoring pass after we merge this PR